modules/builds-configmaps-use-case-consuming-in-env-vars.adoc

```diff
@@ -0,0 +1,84 @@
+// Module included in the following assemblies:
+//
+//* builds/builds-configmaps.adoc
+
+[id="builds-configmaps-use-case-consuming-in-env-vars_{context}"]
+= Consuming ConfigMaps in environment variables
+
+ConfigMaps can be used to populate individual environment variables or to populate environment variables from all keys that form valid environment variable names. As an example, consider the following ConfigMaps:
+
+.ConfigMap with two environment variables
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: special-config <1>
+  namespace: default
+data:
+  special.how: very <2>
+  special.type: charm <2>
+----
+<1> Name of the ConfigMap.
+<2> Environment variables to inject.
+
+.ConfigMap with one environment variable
+[source,yaml]
+----
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: env-config <1>
+  namespace: default
+data:
+  log_level: INFO <2>
+----
+<1> Name of the `ConfigMap`.
+<2> Environment variable to inject.
+
+.Procedure
+
+* You can consume the keys of this ConfigMap in a pod using `configMapKeyRef` sections.
++
+.Sample pod specification configured to inject specific environment variables
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: dapi-test-pod
+spec:
+  containers:
+    - name: test-container
+      image: gcr.io/google_containers/busybox
+      command: [ "/bin/sh", "-c", "env" ]
+      env: <1>
+        - name: SPECIAL_LEVEL_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: special-config <2>
+              key: special.how <3>
+        - name: SPECIAL_TYPE_KEY
+          valueFrom:
+            configMapKeyRef:
+              name: special-config <2>
+              key: special.type <3>
+              optional: true <4>
+      envFrom: <5>
+        - configMapRef:
+            name: env-config <6>
+  restartPolicy: Never
+----
+<1> Stanza to pull the specified environment variables from a ConfigMap.
+<2> Name of the ConfigMap to pull specific environment variables from.
+<3> Environment variable to pull from the ConfigMap.
+<4> Makes the environment variable optional. As optional, the pod will be started even if the specified ConfigMap and keys do not exist.
+<5> Stanza to pull all environment variables from a ConfigMap.
+<6> Name of the ConfigMap to pull all environment variables.
++
+When this pod is run, the pod logs will include the following output:
++
+----
+SPECIAL_LEVEL_KEY=very
```

Comment:
Is 'SPECIAL_TYPE_KEY=charm' not in here because of the `optional: true` entry? If so, I think a note below this code block stating that might be useful to the reader. Or, should that environment variable be instead included in this code block?